src/spikeinterface/qualitymetrics/pca_metrics.py

```diff
@@ -643,7 +643,10 @@ def nearest_neighbors_noise_overlap(
 
     templates_ext = sorting_analyzer.get_extension("templates")
     assert templates_ext is not None, "nearest_neighbors_isolation() need extension 'templates'"
-
+    if sorting_analyzer.has_extension("waveforms"):
+        template_operator = "median"
+    else:
+        template_operator = "mean"
```

Comment:
Not sure to like this. Because the results depend on the sate of analyzer and not on explicit parameters.

Reply:
plus this should be `average`

Reply:
actually the `QualityMetrics` extension depends on `waveforms`, so I'll just revert

Reply:
done